echemdb/data/cv/entry.py

```diff
@@ -100,40 +100,123 @@ def __getitem__(self, name):
         """
         return Descriptor(self.package.descriptor)[name]
 
-    def df(self, yunit=None):
+    def xy_units(self, xunit=None, yunit=None):
+        r"""Returns a set of x and y units as astropy units, where the resulting xunit has the magnitude of 
+        volt (V) and the resulting yunit has the magnitude of a current (I) or a current density (j).
+
+        Whether or not the yunit is in I or j, is determined from the column names of the data frame.
+
+        Examples:
+
+        When the parameters for `xunit` and `yunit` parameters are `None`, SI units are returned.::
+
+            >>> entry = Entry.create_examples()[0]
+            >>> entry.xy_units()
+            (Unit("V"), Unit("A / m2"))
+
+        When the parameter is set to 'original', the original values of the publisged figure are returned.::
+
+            >>> entry = Entry.create_examples()[0]
+            >>> entry.xy_units(xunit='original', yunit='original')
+            (Unit("V"), Unit("mA / cm2"))
+
+        The resulting units can also be set separately.::
+
+            >>> entry = Entry.create_examples()[0]
+            >>> entry.xy_units(xunit='original', yunit='uA / cm2')
+            (Unit("V"), Unit("uA / cm2"))
+
+        """
+        import pandas as pd
+        from astropy import units as u
+
+        column_names = pd.read_csv(self.package.resources[0].raw_iter(stream=False)).columns
+        
+        if yunit is None:
+            if 'j' in column_names:
+                yunit = u.A / u.m**2
+            if 'I' in column_names:
+                yunit = u.A
+        if yunit == 'original':
+            yunit = self.figure_description.current.unit
+        if isinstance(yunit, str):
+            yunit = u.Unit(yunit)
+
+        if xunit is None:
+            xunit = u.V
+        if xunit == 'original':
+            xunit = self.figure_description.potential_scale.unit
+        if isinstance(xunit, str):
+            xunit = u.Unit(xunit)
+
+        return xunit, yunit
+
+    def df(self, xunit=None, yunit=None):
         r"""
         Return the CSV resource attached to this entry as a data frame.
 
-        EXAMPLES::
+        If the x and yunits are not specified the values in the respective 
+        columns are given in SI units. The data frame can also be returned 
+        with the original figure units or with custom units as shown in the following examples.
+
+        EXAMPLES:
+        
+        The first example provides a data frame in SI units.::
```

Comment:
```suggestion
        A data frame in SI units::
```

I would not number things because it makes inserting another example later painful.